packages/nimble-components/src/theme-provider/index.ts

```diff
@@ -33,19 +38,29 @@ export const theme = DesignToken.create<Theme>({
  * @internal
  */
 export class ThemeProvider extends FoundationElement {
-    @attr({
-        attribute: 'direction'
-    })
+    @attr()
+    public override lang: string = document.documentElement.lang;
```

Comment:
I think we might need to explicitly fallback to 'en-US' as the default if this isn't a non-empty string:
- On a page where the `lang` attribute is not present on the `html` element, `document.documentElement.lang` is `''`. It looks like writing `lang=""` is also [allowed (it means a page with undetermined languge)](https://www.w3.org/International/questions/qa-no-language#undetermined)
- If we pass `''` as the locale to `Intl.DateTimeFormat` that yields an error:
    ```
    > new Intl.DateTimeFormat('', { dateStyle: 'medium', timeStyle: 'medium' });
    Uncaught RangeError: Incorrect locale information provided
    ```

I don't think we want our controls to error out if locale is unset, only if someone explicitly set `themeProvider.lang` to something invalid themselves.

(You might want to wait for @rajsite to weigh in before making changes though)

Reply:
Pulling the document.documentElement.lang at whatever random time this property gets evaluated is not sufficient. To respond to whatever the document configuration is we should actually observe it for changes. The patterns for having the theme provider defaults defined by the document are part of https://github.com/ni/nimble/pull/1257

So for now we should do what the other properties do and initialize to a know value that can be overwritten, ie. `en-US`.

I'm also not sure about making every element that wants to use a locale having to do all the validation. Seems like something that should be localized to where the local is configured. This should probably be a stand-up discussion topic or something.

Reply:
We talked and came up with a plan:
- the `lang` token's default value will be whatever the page's `lang` attribute is (using a `MutationObserver` to track the value)
    - if the page's `lang` value is invalid, the token instead defaults to the system's default locale (`(new Intl.DateTimeFormat(undefined)).resolvedOptions().locale`)
- if the theme provider has a `lang` value set on it, that overrides the `lang` token's default.
    - if the theme provider's `lang` value is invalid, a validity flag is set on the theme provider, and the default `lang` token value is used instead.

Note that an "invalid" `lang` value is one that causes the `Intl.DateTimeFormat` constructor to throw a range error. Other seemingly "wrong' values can still resolve to a valid locale based on the resolution algorithm. For example, on my machine, the value "foo" resolves to "en-US". However "" and "123" do not resolve and are thus invalid.